src/librustdoc/html/static/main.js

```diff
@@ -1938,18 +1938,27 @@
         if (collapse) {
             toggleAllDocs(pageId, true);
         }
-        if (getCurrentValue('rustdoc-trait-implementations') !== "false") {
-            onEach(document.getElementsByClassName("collapse-toggle"), function(e) {
+        var collapser = function(e) {
                 // inherent impl ids are like 'impl' or impl-<number>'.
                 // they will never be hidden by default.
-                var n = e.parentNode;
+                var n = e.parentElement;
                 if (n.id.match(/^impl(?:-\d+)?$/) === null) {
                     // Automatically minimize all non-inherent impls
                     if (collapse || hasClass(n, 'impl')) {
                         collapseDocs(e, "hide", pageId);
                     }
                 }
-            });
+        };
+        if (getCurrentValue('rustdoc-trait-implementations') !== "false") {
+            onEach(document.getElementById('implementations-list')
+                           .getElementsByClassName("collapse-toggle"), collapser);
+        }
+        if (getCurrentValue('rustdoc-method-docs') !== "false") {
+            var implItems = document.getElementsByClassName('impl-items');
+
+            if (implItems && implItems.length > 0) {
+                onEach(implItems[0].getElementsByClassName("collapse-toggle"), collapser);
+            }
```

Comment:
Good point! I'll fix it.

Reply:
It actually collapses all elements in `implItems`. So I guess it's all good?

Reply:
It looks like it just accesses `implItems[0]`, though. Does that work out for everything we need it for?

Reply:
That's because we use `class` and not `id`, but there is only one `impl-items` element in the DOM. Since `getElementsByClassName` returns a list, I need to get the first (and only) one.

Reply:
To give more details: the `impl-items` element contains all the others which we want to collapse. Therefore, we loop over those items (their collapser to be more exact) and collapse them all.

Reply:
> but there is only one `impl-items` element in the DOM.

Except that's not true. `.impl-items` is applied to all inherent impl sections and all trait impls. Take [the page for `HashMap`](https://doc.rust-lang.org/1.28.0/std/collections/struct.HashMap.html); because it has multiple inherent impl blocks, each of those sections gets its own `.impl-items` element. And below that, all the trait impls wrap the trait methods in their own `.impl-items` element. We can't assume that this current code will grab everything we want it to, because i can only assume it will fall flat on this page.

Reply:
Isn't it what you wanted? If not it needs to be fixed. :)

Reply:
I'm asking what *you* wanted to do with this code in the first place. I'm not sure if you wanted to fold up only the inherent impl methods or all of them. With no PR description talking about what you wanted to do with the PR, i'm forced to guess from what the code is doing.

Reply:
Ah sorry, thought the description was full enough. So yes, this was what I intended to do.

Reply:
Cool! In that case, we're good to go!